pkg/testsuite/mock/blockissuer.go

```diff
@@ -320,226 +294,129 @@ func (i *BlockIssuer) IssueActivity(ctx context.Context, wg *sync.WaitGroup, sta
-	i.privateKey = otherBlockIssuer.privateKey
-	i.PublicKey = otherBlockIssuer.PublicKey
-	i.AccountID = otherBlockIssuer.AccountID
+	i.keyManager = otherBlockIssuer.keyManager
+	i.AccountData = otherBlockIssuer.AccountData
 	i.Validator = otherBlockIssuer.Validator
 }
 
-// getReferencesWithRetry tries to get references for the given payload. If it fails, it will retry at regular intervals until
-// the timeout is reached.
-func (i *BlockIssuer) getReferencesWithRetry(ctx context.Context, parentsCount int, node *Node) (references model.ParentReferences, err error) {
-	timeout := time.NewTimer(i.optsTipSelectionTimeout)
-	interval := time.NewTicker(i.optsTipSelectionRetryInterval)
-	defer timeutil.CleanupTimer(timeout)
-	defer timeutil.CleanupTicker(interval)
-
-	for {
-		references = node.Protocol.Engines.Main.Get().TipSelection.SelectTips(parentsCount)
-		if len(references[iotago.StrongParentType]) > 0 {
-			return references, nil
-		}
+func (i *BlockIssuer) retrieveAPI(blockParams *BlockHeaderParams) iotago.API {
+	if blockParams.ProtocolVersion != nil {
+		api, err := i.Client.APIForVersion(*blockParams.ProtocolVersion)
+		require.NoError(i.Testing, err)
 
-		select {
-		case <-interval.C:
-			i.events.Error.Trigger(ierrors.Wrap(err, "could not get references"))
-			continue
-		case <-timeout.C:
-			return nil, ierrors.New("timeout while trying to select tips and determine references")
-		case <-ctx.Done():
-			return nil, ierrors.Wrap(ctx.Err(), "context canceled whilst trying to select tips and determine references")
-		}
+		return api
 	}
-}
 
-func WithTipSelectionTimeout(timeout time.Duration) options.Option[BlockIssuer] {
-	return func(i *BlockIssuer) {
-		i.optsTipSelectionTimeout = timeout
-	}
+	// It is crucial to get the API from the issuing time/slot as that defines the version with which the block should be issued.
+	return i.Client.APIForTime(*blockParams.IssuingTime)
 }
 
-func WithTipSelectionRetryInterval(interval time.Duration) options.Option[BlockIssuer] {
-	return func(i *BlockIssuer) {
-		i.optsTipSelectionRetryInterval = interval
-	}
-}
+func (i *BlockIssuer) GetNewBlockIssuanceResponse() *api.IssuanceBlockHeaderResponse {
+	defer i.mutex.Unlock()
+	i.mutex.Lock()
 
-func WithIncompleteBlockAccepted(accepted bool) options.Option[BlockIssuer] {
-	return func(i *BlockIssuer) {
-		i.optsIncompleteBlockAccepted = accepted
-	}
+	i.blockIssuanceResponseUsed = false
+	resp, err := i.Client.BlockIssuance(context.Background())
+	require.NoError(i.Testing, err)
+	i.latestBlockIssuanceResp = resp
+
+	return i.latestBlockIssuanceResp
 }
 
-func WithRateSetterEnabled(enabled bool) options.Option[BlockIssuer] {
-	return func(i *BlockIssuer) {
-		i.optsRateSetterEnabled = enabled
+func (i *BlockIssuer) LatestBlockIssuanceResponse(context context.Context) *api.IssuanceBlockHeaderResponse {
+	defer i.mutex.Unlock()
+	i.mutex.Lock()
+
+	if i.blockIssuanceResponseUsed {
```

Comment:
I don't get why we change it in this function, maybe you can add some comments how the logic of this `blockIssuanceResponseUsed` is supposed to work?